prusti-interface/src/emitter/export_error_emitter.rs

```diff
@@ -0,0 +1,142 @@
+use std::io::Write;
+use rustc_data_structures::sync::{self, Lrc};
+use rustc_errors::{
+    annotate_snippet_emitter_writer::AnnotateSnippetEmitterWriter,
+    emitter::{ColorConfig, Destination, Emitter, EmitterWriter, HumanReadableErrorType},
+    json::JsonEmitter,
+    registry::Registry,
+    Diagnostic, DiagnosticId, Handler,
+};
+use rustc_span::source_map::SourceMap;
+use rustc_session::config;
+
+
+pub struct ExportErrorEmitter {
+    inner: Box<dyn Emitter>,
+}
+
+impl Emitter for ExportErrorEmitter {
+    fn emit_diagnostic(&mut self, diag: &Diagnostic) {
+        self.custom_emit(diag);
+    }
+    fn source_map(&self) -> Option<&Lrc<SourceMap>> {
+        self.inner.as_ref().source_map()
+    }
+}
+
+impl ExportErrorEmitter {
+    pub fn new(
+        sopts: config::Options,
+        registry: rustc_errors::registry::Registry,
+        source_map: Lrc<SourceMap>,
+        emitter_dest: Option<Box<dyn Write + std::marker::Send>>,
+    ) -> Self {
+        Self {
+            inner: build_inner_emitter(&sopts, registry, source_map, emitter_dest),
+        }
+    }
+
+    fn custom_emit(&mut self, diag: &Diagnostic) {
+        if let Some(DiagnosticId::Error(err_code)) = &diag.code {
+            if &err_code == &"E0428"
+                && diag
+                    .message
+                    .iter()
+                    .any(|(err_msg, _)| err_msg.contains("prusti_export_"))
+            {
+                let mut diag = diag.to_owned();
+                for (msg, _) in diag.message.iter_mut() {
+                    let msg_vec: Vec<&str> = msg.split("_").collect();
+                    let len = msg_vec.len();
+                    if len >= 4 {
```

Comment:
This seems like a filter that is extremely specific to the error message. Why split on `_`? Why does it need to have 4 or more elements? If there is no other way, please add a comment here to show an example of the error you are filtering out.